gammapy/data/observation_stats.py

```diff
@@ -0,0 +1,189 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+import astropy.units as u
+from ..stats import Stats
+from gammapy.stats import significance_on_off
+
+__all__ = [
+    'ObservationStats',
+]
+
+
+class ObservationStats(Stats):
+    """Observation statistics.
+
+    Class allowing to summarize observation 
+    (`~gammapy.data.DataStoreObservation) statistics
+
+    Parameters
+    ----------
+    n_on : int
+        number of on events
+    n_off : int
+        number of on events
+    a_on : float
+        number of on events
+    a_off : float
+        number of on events
+    obs_id : int
+        number of on events
+    livetime : float
+        number of on events
+    alpha : float
+        number of on events
+
+    """
+
+    def __init__(self,
+                 n_on=None, n_off=None, a_on=None, a_off=None,
+                 obs_id=None, livetime=None, alpha=None):
+        super(ObservationStats, self).__init__(
+            n_on=n_on,
+            n_off=n_off,
+            a_on=a_on,
+            a_off=a_off
+        )
+
+        self.obs_id = obs_id
+        self.livetime = livetime
+        self.alpha_obs = alpha
+
+    @classmethod
+    def from_target(cls, obs, target, bg_estimate):
+        """
+        Auxiliary constructor from an osbervation, a target
+        a background estimate
+
+        Parameters
+        ----------
+        obs_table : `~gammapy.data.ObservationTable`
+            Observation index table
+        target_pos : `~astropy.coordinates.SkyCoord`
+            Target position
+        bg_method : str
+            Background estimation method
+        """
+        n_on = cls._get_on_events(obs, target)
+        n_off = len(bg_estimate.off_events)
+        a_on = bg_estimate.a_on
+        a_off = bg_estimate.a_off
+
+        obs_id = obs.obs_id
+        livetime = obs.observation_live_time_duration
+        alpha = a_on / a_off
+        stats = cls(n_on=n_on,
+                    n_off=n_off,
+                    a_on=a_on,
+                    a_off=a_off,
+                    obs_id=obs_id,
+                    livetime=livetime,
+                    alpha=alpha)
+        return stats
+
+    @property
+    def alpha(self):
+        """Override member function from `~gammapy.stats.Stats`
+        to take into account weighted alpha by number of Off events
+        """
+        return self.alpha_obs
+
+    @property
+    def sigma(self):
```

Comment:
Ah yes, I forgot about that tricky part for the alpha computation.

Note that at some point I added this standalone function:
http://docs.gammapy.org/en/latest/api/gammapy.stats.combine_stats.html

This has some version of the alpha computation for two observations (probably not debugged / tested).
It should somehow be integrated with the `Stats` class better.
Maybe it's possible to have a `stack` classmethod on the base `Stats` class as well, i.e. follow the design pattern you implemented for `ObservationStats`?

> So I guess we need to re-implement the property alpha in the ObservationStats to deal with the weighting, right ?

I'm not sure. Currently your design is such that the stacked `ObservationStats` doesn't contain arrays of `n_on`, `a_on`, ... but just the stacked values, right? Then the weighting has to happen at stacking time, not alpha computation time, no?

Is it clear to you how to handle / resolve this?
If no, I could have a closer look and think about it, or we delay this issue to a future pull request.
Let me know what you prefer ...